src/validators.ts

```diff
@@ -36,6 +45,16 @@ export function findSeatPrice(validators: (CurrentEpochValidatorInfo | NextEpoch
     return left;
 }
 
+function findSeatPriceForProtocolAfter49(validators: (CurrentEpochValidatorInfo | NextEpochValidatorInfo)[], maxNumberOfSeats: number, minimumStakeRatio: number): BN {
+    const stakes = validators.map(v => new BN(v.stake, 10)).sort((a, b) => a.cmp(b));
+    const stakesSum = stakes.reduce((a, b) => a.add(b));
+    if (validators.length < maxNumberOfSeats) {
+        return stakesSum.div(new BN(minimumStakeRatio));
+    } else {
+        return stakes[0].add(new BN(1));
+    }
+}
```

Comment:
LGTM, but let's have a link to nearcore implementation for the future reference: https://github.com/near/nearcore/blob/5a8ae263ec07930cd34d0dcf5bcee250c67c02aa/chain/epoch_manager/src/validator_selection.rs#L308;L315